lib/datadog/core/telemetry/logging.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module Datadog
+  module Core
+    module Telemetry
+      # Logging interface for sending telemetry logs
+      module Logging
+        module_function
+
+        def report(exception, level:)
+          # Annoymous exceptions to be logged as <Class:0x00007f8b1c0b3b40>
+          message = exception.class.name || exception.class.inspect
+
+          event = Event::Log.new(
+            message: message,
+            level: level
+          )
+
+          if (telemetry = Datadog.send(:components).telemetry)
+            telemetry.log!(event)
+          else
+            Datadog.logger.debug do
+              "Sending telemetry log about #{message} before telemetry component is ready"
```

Comment:
I find "sending" to be misleading because nothing is actually sent. Suggest either "attempting to send" or "requested to send" or similar. Also, I would put the message at the end so that it's clear where the diagnostics of this component ends and where the forwarded message is. E.g. "Requested to send telemetry log when telemetry component is not ready: #{message}". Is it possible for telemetry to be never turned on, i.e. it would actually not ever become ready?

Reply:
The telemetry object is always created, even if telemetry is disabled https://github.com/DataDog/dd-trace-rb/blob/b7b99b2c72ca97cb42b7c8b72accb5571e8c4f3f/lib/datadog/core/telemetry/component.rb#L45-L47
so it will eventually become no-nil.

I also like that this message is at `debug` level, so people have to manually enable debug-level logs to see it.

I do agree that the messaging can be more precise.

Reply:
Minor: Should we perhaps print the whole `event` and not just the `message` when logging

Reply:
I don't think printing the whole event is necessary, since the issue is not caused by event itself.

Reply:
My thinking for this suggestion is, if later we add more information to the event (e.g. stack), we'll need to remember to tweak the debug message, whereas if we print the whole event, then we won't need to touch it again.